micrometer-core/src/main/java/io/micrometer/core/instrument/binder/grpc/GrpcClientObservationContext.java

```diff
@@ -32,110 +32,77 @@
  */
 public class GrpcClientObservationContext extends RequestReplySenderContext<Metadata, Object> {
 
-    private String serviceName;
-
-    private String methodName;
-
-    private String fullMethodName;
-
-    private MethodType methodType;
-
-    @Nullable
-    private Code statusCode;
-
-    private String authority;
-
-    private Metadata headers;
-
-    private Metadata trailers;
+    private GrpcObservationContextInfo grpcInfo;
```

Comment:
I think in this particular case having a common `GrpcObservationContext` might be a better solution.

Also, I don't thin the automated refactoring too does what you claim:
>reducing complexity and enhancing readability of your source code.

I think this code is more complex and less readable than it was before. _In some cases_, if we need to change things, this could be an improvement.

Also now that there is an extra object inside of this class that is heavily used for very short-lived object, introducing this could also mean worse performance (didn't measure).